### ISSUE TITLE
CMakeLists.txt: unable to set PUGIXML_WCHAR_MODE

### ISSUE THREAD

**Issue by gnammix** [zeux/pugixml#413]
Hello, 

CMakeLists.txt dropped processing the BUILD_SETTING variable that, afaik, was the only way to set PUGIXML_WCHAR_MODE when build the library in non header only mode.

This was dropped:
```
if(DEFINED BUILD_DEFINES)
	foreach(DEFINE ${BUILD_DEFINES})
		add_definitions("-D" ${DEFINE})
	endforeach()
endif()
```
for
```
set(BUILD_DEFINES CACHE STRING "Build defines")
```

Possibly I'm missing something and there is a proper way to do it :)

Tested on WIndows VS2019 with cmake version 3.19.20122902-MSVC_2

**Comment by zeux** [comment]
I thought you can override options declared via `set` externally? Maybe I'm wrong, I'm not a CMake expert. cc @slurps-mad-rips 

Somewhat related to #388 although I'm not sure if the requested feedback on that change will be implemented.

**Comment by slurps-mad-rips** [comment]
> I thought you can override options declared via `set` externally?

Correct. Setting `PUGIXML_WCHAR_MODE` with either `-DPUGIXML_WCHAR_MODE=YES` when configuring CMake *or* calling `set(PUGIXML_WCHAR_MODE YES)` before using `add_subdirectory` will work.

**Comment by slurps-mad-rips** [comment]
Actually looking into it, the `BUILD_DEFINES` was never updated.

A PR with

```cmake
add_compile_definitions(${BUILD_DEFINES})
```

is needed. Luckily it's just the one line, though I am wondering why this wasn't caught before, or maybe it was removed in some rebase/merge 😞

**Comment by zeux** [comment]
> Actually looking into it, the BUILD_DEFINES was never updated.

Ah, that explains it, thanks :)

**Comment by bjornblissing** [comment]
> I am wondering why this wasn't caught before, or maybe it was removed in some rebase/merge 😞
> 

The commit which removed `BUILD_DEFINES` can be found here: 1c5a0bb

**Comment by bjornblissing** [comment]
I have re-introduced the `BUILD_DEFINES` functionallity in PR #399. However, the way of handling the available options suggested by #388 is probably a better solution in the long run.

**Comment by zeux** [comment]
Fixed by #399 (and #388 will make this easier in the future).
